docs/design/variadics.md

```diff
@@ -0,0 +1,1018 @@
+form will likewise be well-shaped), but need not be well-typed. This enables us
+to apply these reductions while determining whether an utterance is well-typed,
+as in the case of typing an expression or pattern that contains a pack literal
+or arity coercion, above.
+
+_Singular pack removal:_ if `E` is a pack segment, `⟬E⟭` reduces to `E`.
+
+_Singular expansion removal:_ `...E` reduces to `E`, if `E` contains no pack
+literals, arity coercions, or each-names.
+
+_Pack expansion splitting:_ If `E` is a segment and `S` is a sequence of
+segments, `...⟬E, S⟭` reduces to `...E, ...⟬S⟭`.
+
+_Pack expanding:_ If `F` is a function, `X` is an utterance that does not
+contain pack literals or each-names, and `⟬P1, P2⟭` and `⟬Q1, Q2⟭` both have the
+shape `(S1, S2)`, then `F(⟬P1, P2⟭, X, ⟬Q1, Q2⟭, ⟪Y; S1+S2⟫)` reduces to
+`⟬F(P1, X, Q1, ⟪Y; S1⟫), F(P2, X, Q2, ⟪Y; S2⟫)⟭`. This rule generalizes in
+several dimensions:
+
+-   `F` can have any number of non-pack-literal arguments, and any positive
+    number of pack literal arguments, and they can be in any order.
+-   The pack literal arguments can have any number of segments (but the
+    well-shapedness requirement means they must have the same number of
+    segments).
+-   `F()` can be any expression syntax other than `...`, not just a function
+    call. For example, this rule implies that `⟬X1, X2⟭ * ⟬Y1, Y2⟭` reduces to
+    `⟬X1 * Y1, X2 * Y2⟭`, where the `*` operator plays the role of `F`.
+-   `F()` can also a be a pattern syntax. For example, this rule implies that
+    `(⟬x1: X1, x2: X2⟭, ⟬y1: Y1, y2: Y2⟭)` reduces to
+    `⟬(x1: X1, y1: Y1), (x2: X2, y2: Y2)⟭`, where the tuple syntax `( , )` plays
+    the role of `F`.
+
+_Coercion expanding:_ If `F` is a function, `S` is a shape, and `Y` is an
+expression that does not contain pack literals or arity coercions,
+`F(⟪X; S⟫, Y, ⟪Z; S⟫)` reduces to `⟪F(X, Y, Z); S⟫`. As with pack expanding,
+this rule generalizes:
+
+-   `F` can have any number of non-arity-coercion arguments, and any positive
+    number of arity coercion arguments, and they can be in any order.
+-   `F()` can be any expression syntax other than `...` or pack literal
+    formation, not just a function call. Unlike pack expanding, coercion
+    expanding does not apply if `F` is a pattern syntax.
+
+_Coercion removal:_ `⟪E; 1⟫` reduces to `E`.
+
+_Tuple indexing:_ Let `I` be an integer template constant, let `X` be a tuple
+segment, and let `Ys` be a sequence of tuple segments.
+
+-   If the arity `A` of `X` is less than `I+1`, then `(X, Ys).(I)` reduces to
+    `(Ys).(I-A)`.
+-   Otherwise:
+    -   If `X` is not a pack expansion, then `(X, Ys).(I)` reduces to `X`.
+    -   If `X` is of the form `...⟬⟪E; S⟫⟭`, then `(X, Ys).(I)` reduces to `E`.
+
+### Other equivalences
+
+Unless otherwise specified, all expressions in these rules must be free of side
+effects.
+
+_Coercion merging:_
+
+-   `⟪E; M⟫, ⟪E; N⟫` is equivalent to `⟪E; M, N⟫`.
```

Comment:
Shouldn't you only have `+` to the right of a `;` in a `⟪E; ...⟫`? Having both `+` and `,` means there are a lot more rules than would otherwise be needed. And a lot of the rules are just what was already stated about arities.